docs/howtos/vpn.md

```diff
@@ -61,92 +58,77 @@ Using the nsenter command:
 ```
 
-With that setup, the openvpn container can be restarted when needed, and
-the other environment will recover the connectivity transparently.
+Using podenv namespace, the openvpn container can be restarted when needed, and
+the other applications will recover the connectivity transparently.
 
 
 ## Persist the configuration
 
-Using command line arguments to manage network can become difficult.
-To simplify podenv usage, it's better to specify the configuration
-of commonly used environment.
+Using command line arguments to manage network can be difficult.
+To simplify podenv usage, it is recommended to create a static configuration.
 
-Here is an example configuration for such setup:
+Write this configuration file in `~/.config/podenv/corp.dhall`:
 
 ```dhall
-let Podenv = env:PODENV_PRELUDE
-
-let Hub = env:PODENV_HUB
-
-let vpn-envs =
-      [ Hub.Environments.OpenVPN // { home = Some "~/.config/openvpn" }
-      , Hub.Environments.Firefox.Extended
-      , Hub.Environments.Shell
-      ]
-
-let work-envs =
-      Hub.Functions.mapEnv
-        (     \(env : Podenv.Env.Type)
-          ->  Hub.Runtimes.Fedora.Create.Latest
-                (env // { name = "work-" ++ env.name, network = Some "work" })
-        )
-
-in Hub.Defaults # work-envs vpn-envs
+-- | Load the podenv/hub configuration
+let Hub = ./Hub.dhall
+
+-- | An application setting for the namespace
+let ns = { namespace = Some "corp-vpn" }
+
+-- | The resulting applications tree
+in {
+  vpn = Hub.openvpn         // { volumes = [ "~/.config/openvpn:~/" ] }    // ns,
+  web = Hub.firefox.default // { volumes = [ "~/.config/firefox-vpn:~" ] } // ns,
+  bridge = Hub.ssh.client "user@internal-host"                             // ns
+}
 ```
 
-The statements mean:
+Then add the applications tree to the main configuration in `~/.config/podenv/config.dhall`:
 
-* Podenv is the provided types and Hub is the podenv/hub package
-* vpn-envs is a list of environment we want to use inside our vpn
-* work-envs is a function that will apply a function to each environments using the `mapEnv` function.
-* work-envs is applied to vpn-envs and the results is added to the list of environments.
-
-work-envs does three things to it's parameter:
-
-* Setup a Fedora runtime
-* Change the environment name to be prefixed by "work-"
-* Configure the network to be "work"
+```dhall
+./Hub.dhall // { corp = ./corp.dhall }
+```
 
-We now have three new environments:
+Three new applications are now configured to share the namespace:
 
-```console
+```ShellSession
 $ podenv --list
-NAME                 DESCRIPTION
-[...]
-work-firefox         Mozilla Firefox
-work-openvpn         VPN solutions
-work-shell           A simple terminal
+corp.bridge: Application (OpenSSH client)
+corp.vpn: Application (VPN solution)
+corp.web: Application (Mozilla Firefox)
 ```
 
-That we can use like so:
+That can be used like so:
 
-```console
-$ podenv work-openvpn ovpn.conf
+```ShellSession
+$ podenv corp.vpn ovpn.conf
```

Comment:
How to configure the corp.dhall to avoid to pass the openvpn config file ?

Reply:
The command can be set in the new application, but I left that out because it is convenient to be able to select different configs (e.g. to pick a remote), and setting the command would make the example a bit more complicated.